packages/front-end/components/Settings/EditDataSource/DataSourceJupypterQuery/DataSourceJupyterNotebookQuery.tsx

```diff
@@ -1,19 +1,22 @@
 import React, { FC, useCallback, useState } from "react";
 import { FaPencilAlt, FaPlus } from "react-icons/fa";
-import Code from "@/components/SyntaxHighlighting/Code";
-import usePermissions from "@/hooks/usePermissions";
 import { DataSourceQueryEditingModalBaseProps } from "../types";
+import Code from "../../../SyntaxHighlighting/Code";
+import usePermissions from "../../../../hooks/usePermissions";
+import { useDefinitions } from "../../../../services/DefinitionsContext";
 import { EditJupyterNotebookQueryRunner } from "./EditJupyterNotebookQueryRunner";
 
 type DataSourceJupyterNotebookQueryProps = DataSourceQueryEditingModalBaseProps;
 
 export const DataSourceJupyterNotebookQuery: FC<DataSourceJupyterNotebookQueryProps> = ({
   onSave,
   dataSource,
+  canEdit = true,
 }) => {
+  const { project } = useDefinitions();
   const [uiMode, setUiMode] = useState<"view" | "edit">("view");
   const permissions = usePermissions();
-  const canEdit = permissions.editDatasourceSettings;
+  canEdit = canEdit && permissions.check("editDatasourceSettings", project);
```

Comment:
This should use the datasource's projects, not the project set in definitions

Reply:
same update as above